### DISCUSSION TITLE
Jamulus Usage Public Dataset

### DISCUSSION THREAD

**Discussion by dtinth** [jamulussoftware/jamulus#3286] [General]
In Thailand Jamulus community, I maintain [a web page that lists the top Jamulus users in Thailand](https://mjth.live/community/stats/):

<img width="484" alt="image" src="https://github.com/jamulussoftware/jamulus/assets/193136/1292347e-8973-495c-9751-e8d234959d9a">

I was able to generate the report by collecting snapshots of [explorer.jamulus.io](https://explorer.jamulus.io/) every few minutes[^1]. With that, I have a data lake of publicly-available Jamulus information. Then with some data warehousing, I can ask questions about the data I have.

[^1]: I run instances of [Jamulus explorer backend](https://github.com/dtinth/jamulus-php) on my own infrastructure, so that the main Jamulus explorer is not impacted by this project.

I recently upgraded the infrastructure behind this project (moving from a SQLite database to Google BigQuery), and in doing so, I was able to make the dataset public, so anyone interested can run SQL queries against this dataset:

<img width="1256" alt="image" src="https://github.com/jamulussoftware/jamulus/assets/193136/38b715ac-403c-4851-bb77-350ea9ca3419">

The dataset contains information about all public servers, aggregated at hourly interval. I have been collecting this data since 2023-08-01. With this dataset, I can ask questions about various things. For example,

- “Which hour of the day I (dtinth) use Jamulus the most in 2024?”

  <img width="775" alt="image" src="https://github.com/jamulussoftware/jamulus/assets/193136/0bd463e0-d72a-4109-b94e-d4fdb8c8093f">

- “Which countries spent most time combined in Jamulus in May 2024?”

  <img width="705" alt="image" src="https://github.com/jamulussoftware/jamulus/assets/193136/6b70f308-6201-41dd-ac3d-2ab805535227">

Each row in the dataset represents a single client in a single server in a given hour. For example, this row:

```json
{
  "date": "2024-05-03",
  "hour": "16",
  "hours_seen": "0.36",
  "client_name": "dtinth",
  "client_country": "Thailand",
  "client_city": "Bangkok",
  "client_instrument": "Keyboard",
  "client_skill": "Beginner",
  "server_name": "🏘 MJTH.live",
  "server_country": "Thailand",
  "server_city": "Z.com Cloud, BKK",
  "server_ip": "150.95.25.226",
  "server_port": "22124",
  "server_directory_name": "Any Genre 1"
}
```

Means that on 2024-05-03, 16:00 ~ 16:59 UTC (11pm local time), I (dtinth) connected to the "🏘 MJTH.live" server and was seen in there for 22 minutes.

If you're interested in querying this dataset, you can try running this in [BigQuery Studio](https://console.cloud.google.com/bigquery):

```sql
SELECT *
FROM `dtinth-storage-space.jamulus.clients`
ORDER BY date DESC, hour DESC
LIMIT 100
```

(More info available at <https://github.com/dtinth/jamulus-archive>)

**Comment by mcfnord** [comment]
I have a similar dataset that is about a year old. I hope we can detect growth trends by nation. A sensitive metric could help us deploy servers where interest is rising. I hope we find ways data like this can help our users understand and navigate their options.

**Comment by ann0see** [comment]
Thanks for sharing. Probably gathering this information is valuable to gain insights on usage. What do you use it for or is it just "nice to know"?

Just as side note, we may want to clarify the privacy policy with examples like this as users should be aware of data collection. 

https://jamulus.io/wiki/Privacy-Statement

**Reply by mcfnord** [reply]
We should clarify that servers on public directories expose details that are logged by third parties. We say unmodified servers don't log or store data, but in reality, I know every name that played a dulcimer in the last year. Linking our privacy policy to these datasets could help. Some server operators block third-party data collection, but it requires an extra step to opt out, and we haven't adopted a standard for signaling participation wishes. I think this kind of data aggregation can lead to powerful tools, but nonetheless meticulously surveils our pseudo-anonymous community.

**Reply by ann0see** [reply]
> that are logged by third parties.

-> Can be logged

We don't know if logging happens this second. But we know that it can happen to the extent already described.

**Reply by dtinth** [reply]
@ann0see 

> Probably gathering this information is valuable to gain insights on usage. What do you use it for or is it just "nice to know"?

Mostly it’s nice-to-know. But after each month/year I usually post in our Discord to celebrate the top jammers in the community.

**Reply by ann0see** [reply]
Thanks! So it's mainly for community building and "gamification".
The Thai community around you is quite active, I think. So it makes sense.
